video_store.py

```diff
@@ -0,0 +1,156 @@
+import requests 
+import datetime 
+
+class Videostore:
+    def __init__(self, url="http://localhost:5000", selected_video=None, selected_customer=None):
+        self.url = url
+        self.selected_video = selected_video
+        self.selected_customer = selected_customer 
+
+# ==========All For Video Options========== 
+
+    # Option 1: Add a video 
+    def create_video(self,title="default", release_date="default", total_inventory=None):
+        query_params = {
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory
+        }
+        response = requests.post(self.url+"/videos",json=query_params)
+        return response.json()
+
+    # Option 2: Edit a video 
+    def update_video(self,video_id="default", title="title", release_date="default", total_inventory=None):
+        if not title:
+            title = self.selected_video["title"]
+        if not release_date:
+            release_date = self.selected_video["release_date"]
+        if not total_inventory:
+            total_inventory = self.selected_video["total_inventory"]
+
+        query_params = {
+        "title": title,
+        "release_date": release_date,
+        "total_inventory": total_inventory
+        }
+
+        response = requests.put(
+            self.url+f"/videos/{video_id}", 
+            json=query_params
+            )
+        # print("response:", response)
+        self.selected_video = response.json() 
+        return response.json()
+
+    # Option 3: Delete a video 
+    def delete_video(self):
+        response = requests.delete(self.url+f"/videos/{id}") 
+        return response.json()
+
+    # Option 4: Get information about all videos
+    def list_videos(self):
+        response = requests.get(self.url+"/videos")
+        return response.json()
+
+    # Option 5: Get information of one video 
+    def get_video(self, title="title", id="default"):
+        
+        for video in self.get_video():
+            if title:
+                if video["title"]==title:
+                    title = video["title"]
+                    self.selected_video = video
+            elif id == video["id"]:
+                self.selected_task = video
+
+        if self.selected_video == None:
+            return "Could not find video by that name or id"
+
+        response = requests.get(self.url+f"/videos/{id}")
+        return response.json()
+
+# ==========All For Customers Options========== 
+
+    # Option 6: Add a customer 
+    def create_customer(self,name="name",postal_code="postal_code",phone="phone"):
+        query_params = {
+            "name": name,
+            "postal_code": postal_code,
+            "phone": phone
+        }
+        response = requests.post(self.url+"/customers",json=query_params)
+        return response.json()
```

Comment:
One opportunity for improvement for many people for this project is handling error codes from the API requests. In general, when writing code to call an API, it's appropriate to check whether the request was successful or it triggered some sort of error. This is where HTTP response codes come in handy! We can usually assume that a 2XX response code means there was no error and any other response code means there was some sort of error. It's very common to have an if...elif...else chain to handle different ranges of response codes. 

This applies to all the other API calls as well.